test/credentials_test.dart

```diff
@@ -147,24 +144,87 @@ Future<void> main([List<String>? args]) async {
       final authProvider = EmailPasswordAuthProvider(app);
-      await authProvider.registerUser(username, password);
+      await authProvider.registerUser(_validUsername, _strongPassword);
     }, appName: "emailConfirm", skip: "It is a manual test");
 
     baasTest('2. Take the recieved token from the email and confirm the user', (configuration) async {
-      //Enter valid token and tokenId from the received email
+      // Enter valid token and tokenId from the received email
       String token = "3a8bdfa28e147f38e531cf5aca93d452a11efc4fc9a81f00219b0cb29cfb93858f6b174123659a6ef47b58a2b80eac3b406d7803605c17ef44401ec6cf2c8fa6";
       String tokenId = "626934dcb4e7e5a0e2f1d85e";
 
       final app = App(configuration);
       final authProvider = EmailPasswordAuthProvider(app);
       await authProvider.confirmUser(token, tokenId);
-      final user = await retryLogin(3, app.logIn, Credentials.emailPassword(username, password));
+      final user = await retryLogin(3, app.logIn, Credentials.emailPassword(_validUsername, _strongPassword));
+      expect(user, isNotNull);
+    }, appName: "emailConfirm", skip: "Run this test manually after test 1 and after setting token and tokenId");
+  });
+
+  baasTest('Email/Password - retry custom confirmation function', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "realm_tests_pending_confirm_${generateRandomString(5)}@bar.com";
+    await authProvider.registerUser(username, _strongPassword);
+
+    await authProvider.retryCustomConfirmationFunction(username);
+
+    final user = await retryLogin(3, app.logIn, Credentials.emailPassword(username, _strongPassword));
+    expect(user, isNotNull);
+  }, appName: "flexible");
+
+  baasTest('Email/Password - retry custom confirmation after user is confirmed', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "realm_tests_do_autoverify_${generateRandomString(5)}@bar.com";
+    // Custom confirmation function confirms automatically username with 'realm_tests_do_autoverify'.
+    await authProvider.registerUser(username, _strongPassword);
+
+    expect(() async {
+      await authProvider.retryCustomConfirmationFunction(username);
+    }, throws<RealmException>("already confirmed"));
+  });
+
+  baasTest('Email/Password - retry custom confirmation for not registered user', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "${generateRandomString(5)}@bar.com";
+    expect(() async {
+      await authProvider.retryCustomConfirmationFunction(username);
+    }, throws<RealmException>("user not found"));
+  });
+
+  group("Email/Password - resend confirm - manual tests", () {
+    // The tests in this group are for manual testing, since they require interaction with mail box.
```

Comment:
as commented on another PR I suggest we move these comments on the `group` call. And name the group `Manual` and the tests with `Manual test`